object_store/src/aws/builder.rs

```diff
@@ -824,6 +827,12 @@ impl AmazonS3Builder {
         self
     }
 
+    /// Set the number of permits for the semaphore controlling concurrency.
+    pub fn with_sempahore_permits(mut self, permits: usize) -> Self {
```

Comment:
```suggestion
    /// Limit the maximum number of concurrent HTTP requests.
    pub fn with_concurrency_limit(mut self, permits: usize) -> Self {
```

I would also suggest moving this onto ClientOptions